platform/commonUI/browse/src/navigation/NavigateAction.js

```diff
@@ -48,20 +48,34 @@ define(
          */
         NavigateAction.prototype.perform = function () {
             var self = this,
-                navigationAllowed = true;
+                navigateTo = this.domainObject,
+                currentObject = self.navigationService.getNavigation(),
+                editing = currentObject.hasCapability('editor') &&
```

Comment:
I _think_ this variable assignment could be moved into `cancelIfEditing` to simplify readability of that function a bit